addon/PlacesProvider.js

```diff
@@ -420,21 +402,42 @@ Links.prototype = {
     let links = yield this.executePlacesQuery(sqlQuery, {
       columns: [
         "bookmarkGuid",
-        "favicon",
         "frecency",
         "guid",
         "lastVisitDate",
-        "mimeType",
         "title",
         "type",
         "url"
       ],
       params: {limit}
     });
 
+    links = yield this._getFavicons(links);
     return this._processLinks(links);
   }),
 
+  _getFavicons: Task.async(function*(links) {
+    if (links.length) {
+      let faviconPromises = [];
+      links.forEach(item => {
+        faviconPromises.push(new Promise(resolve => {
+          PlacesUtils.favicons.getFaviconDataForPage(NetUtil.newURI(item.url), (iconuri, len, data, mime) => {
```

Comment:
Thinking about the failure cases here.. potentially `getFaviconDataForPage` could throw or `newURI` doesn't like the url passed in (although this seems unlikely that Places would add a bad url..?). It would lead to a rejection in one of these `Promise`s. The value of `Promise.all` would be the rejection but we're not looking at the result of that anyway, so the main issue is that the first `reject` causes `all` to reject even with other pending requests resulting in `links` being returned when `favicon`s aren't fully added.